Optimize anchorset and flex tree traversal (#22717)
microsoft/FluidFramework: .changeset/spicy-candles-smash.md

---
"@fluidframework/tree": minor
"fluid-framework": minor
---
---
"section": tree
---

> Optimize non-leaf field field access

Reviewer: ```suggestion
Optimize non-leaf field access
```

> Reading of non-leaf children which have previously been read are recovered from cache faster.

Reviewer: ```suggestion
When reading non-leaf children which have been read previously, they are retrieved from cache faster.
```
